### PR TITLE
Keycloak deployment in Docker Swarm

### LINKED ISSUE #59: 1.1 Keycloak deployment in Docker swarm
- Bringing up keycloak through use of [keycloak bitnami image](https://github.com/bitnami/bitnami-docker-keycloak)  , version 14.0.0
- Use [bitnami postgres image](https://github.com/bitnami/bitnami-docker-postgresql) and connect the keycloak with the postgres. Can be tested , by seeing if the keycloak related tables are created.
-  Migrate current auth-cred-db stack file to postgres stack file with image and other config related changes.
- Bring a nginx proxy in-front of keycloak and test if its working. Set up reverse proxy, ssl, rate limits , refer [ keycloak official docs](https://www.keycloak.org/docs/latest/server_installation/#_setting-up-a-load-balancer-or-proxy), [blog](https://itnext.io/nginx-as-reverse-proxy-in-front-of-keycloak-21e4b3f8ec53)
   -  Verifying the configuration https://www.keycloak.org/docs/latest/server_installation/#verifying-the-configuration
- Automate realm creation using imports. Create a realm , configure with general settings and export it and then import that in deployment and need to configure minimal things only like client secret, smtp/optional things.

### PR DESCRIPTION
**Pull request by hackcoderr**
closes #59

### CODE REVIEW

**Comment by abhi4578** [comment]
- Please add only key cloak related files in commits to the PR.

**Review by abhi4578 — COMMENTED** [review]
- In docker swarm,  a standalone keycloak is deployed and clustered. Can removed clustered setup.

**Comment by hackcoderr** [comment]
> * Please add only key cloak related files in commits to the PR.  i.e. those that will belong to single/keycloak folder https://github.com/hackcoderr/iudx-deployment/tree/keycloak/Docker-Swarm-deployment/single-node/keycloak. You can remove all other things - auth, file-server installation docs @hackcoderr

@abhi4578  I m working on keycloak problem only and have gone through your all comments.

File: Docker-Swarm-deployment/single-node/keycloak/keycloak.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+version: "2"
```
[the remark this answers is not in the document]
Comment (abhi4578) [reply Docker-Swarm-deployment/single-node/keycloak/keycloak.yml:1]: https://github.com/datakaveri/iudx-deployment/tree/master/Docker-Swarm-deployment/single-node/auth-cred-db

File: Docker-Swarm-deployment/single-node/keycloak/keycloak-stack.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      - POSTGRESQL_DATABASE=bitnami_keycloak
+    volumes:
+      - "postgresql_data:/bitnami/postgresql"
+  keycloak:
+    image: docker.io/bitnami/keycloak:12-debian-10
+    ports:
+      - "80:8080"
+    environment:
+      - KEYCLOAK_CREATE_ADMIN_USER=true
+    depends_on:
```
Comment (abhi4578) [inline Docker-Swarm-deployment/single-node/keycloak/keycloak-stack.yml:24]: depends on does not work in stack file/ swarm deployment. 

How is the keycloak connecting to postgres ? And also reuse postgres  stack filee defined in  single-node/postgres folder

**Review by abhi4578 — CHANGES_REQUESTED** [review]

**Comment by abhi4578** [comment]
This is the next step :

**Comment by abhi4578** [comment]
This is the next step
- Bring a nginx proxy in-front of keycloak and test if its working. Set up reverse proxy, ssl, rate limits , refer [ keycloak official docs](https://www.keycloak.org/docs/latest/server_installation/#_setting-up-a-load-balancer-or-proxy), [blog](https://itnext.io/nginx-as-reverse-proxy-in-front-of-keycloak-21e4b3f8ec53)
   -  Verifying the configuration https://www.keycloak.org/docs/latest/server_installation/#verifying-the-configuration

Its also mentioned here.https://github.com/datakaveri/iudx-deployment/issues/59

**Comment by abhi4578** [comment]
This is the next step

    Bring a nginx proxy in-front of keycloak and test if its working. Set up reverse proxy, ssl, rate limits , refer keycloak official docs, blog
        Verifying the configuration https://www.keycloak.org/docs/latest/server_installation/#verifying-the-configuration

Its also mentioned here.#59 @hackcoderr

**Comment by abhi4578** [comment]
- keycloak bitnami image , version 14.0.0
- nginx 1.20

**Comment by abhi4578** [comment]
closes issue #59

**Review by abhi4578 — APPROVED** [review]

**Pull request merged**